src/Backups/RestoreCoordinationLocal.cpp

```diff
@@ -51,6 +52,40 @@ bool RestoreCoordinationLocal::acquireReplicatedSQLObjects(const String &, UserD
     return true;
 }
 
+void RestoreCoordinationLocal::generateUUIDForTable(ASTCreateQuery & create_query)
+{
+    String query_str = serializeAST(create_query);
+
+    auto find_in_map = [&]
+    {
+        auto it = create_query_uuids.find(query_str);
+        if (it != create_query_uuids.end())
+        {
+            create_query.setUUID(it->second);
+            return true;
+        }
+        return false;
+    };
+
+    {
+        std::lock_guard lock{mutex};
+        if (find_in_map())
+            return;
+    }
+
+    create_query.setUUID({});
+    auto new_uuids = create_query.generateRandomUUID();
+    String new_query_str = serializeAST(create_query);
+
+    {
+        std::lock_guard lock{mutex};
+        if (find_in_map())
+            return;
+        create_query_uuids[query_str] = new_uuids;
+        create_query_uuids[new_query_str] = new_uuids;
```

Comment:
Why do we need to store both? we can store only the serialized AST, and always use serialized AST during checking right ?. maybe I am missing something?